src/app/shared/components/Breadcrumbs/BreadcrumbsItem/BreadcrumbsItem.tsx

```diff
@@ -84,31 +107,222 @@ const BreadcrumbsItem = (props: BreadcrumbsItemProps): JSX.Element => {
+  };
+
+  const confirmNameChange = () => {
+    const item = currentFolder[0];
+    const metadata: DriveFolderMetadataPayload = { itemName: dirtyName };
+    if (item.name !== dirtyName) {
+      dispatch(storageThunks.updateItemMetadataThunk({ item, metadata }));
+    }
+    setIsEditingName(false);
+  };
+
+  const onNameEnterKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      confirmNameChange();
+    } else if (e.key === 'Escape') {
+      onNameBlurred();
+    }
+  };
+
   return (
-    <div
-      ref={drop}
-      className={`max-w-fit flex ${
-        props.item.isFirstPath ?? 'flex-1'
-      } cursor-pointer flex-row items-center truncate p-1 font-medium ${isDraggingOverClassNames} 
+    <>
+      {!props.item.active && !props.item.dialog ? (
+        <div className="relative flex items-center">
+          <Menu>
+            {isEditingName ? (
+              <input
+                className="dense no-ring rect select-text border border-white"
+                onClick={(e) => e.stopPropagation()}
+                onDoubleClick={(e) => e.stopPropagation()}
+                type="text"
+                value={dirtyName}
+                placeholder={dirtyName}
+                onChange={onNameChanged}
+                onBlur={onNameBlurred}
+                onKeyDown={onNameEnterKeyDown}
+                autoFocus
+                name="fileName"
+              />
+            ) : (
+              <Menu.Button
+                className={'flex cursor-pointer items-center rounded-md p-1 hover:bg-gray-5 active:bg-gray-5'}
+              >
+                <div className="flex items-center">
+                  {dirtyName}
+                  <CaretDown weight="bold" className="ml-1 h-3 w-3" color="#3A3A3B" />
+                </div>
+              </Menu.Button>
+            )}
+            <Transition
+              className={'absolute left-0'}
+              enter="transform transition duration-50 ease-out"
+              enterFrom="scale-98 opacity-0"
+              enterTo="scale-100 opacity-100"
+              leave="transform transition duration-50 ease-out"
+              leaveFrom="scale-98 opacity-100"
+              leaveTo="scale-100 opacity-0"
+            >
+              <Menu.Items
+                className={
+                  'absolute mt-6 w-56 rounded-md border border-black border-opacity-8 bg-white py-1.5 drop-shadow'
+                }
+              >
+                <Menu.Item>
+                  <div
+                    onClick={onCreateFolderButtonClicked}
+                    className="flex cursor-pointer items-center py-2 px-3 text-gray-80 hover:bg-gray-5 active:bg-gray-10"
+                  >
+                    <FolderSimplePlus size={20} />
+                    <p className="ml-3">{i18n.get('actions.upload.folder')}</p>
+                  </div>
+                </Menu.Item>
+                <div className="my-0.5 mx-3 border-t border-gray-10" />
+                {!isBreadcrumbItemShared ? (
```

Comment:
Sharing options (🔴 "Share" and "Get link") should not exclude each other, they are different options.

**Options:**
- "Share" → Always (**🔴 [THIS FEATURE](https://www.figma.com/file/62hpKnJYlxUN77HUECXAlF/Drive-Web?node-id=4290%3A31537&t=PJrB4U0EO0ShnTVU-11) IS NOT YET IMPLEMENTED:** It's another feature different to shared links, don't add it yet).
- "Get link" → Only if folder has not a shared link yet.
- "Copy link", "Link settings" and "Delete link" → When the folder has a shared link.

If you have any question about this, please ask Álvaro.

Reply:
Please re-check this @aboredvaro as we want to merge before this week finishesPlease re-check this @aboredvaro as we want to merge before this week finishesPlease re-check this @aboredvaro as we want to merge before this week finishes

Reply:
Working as expected now 👌🏻